camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/creation/BrooklynEntityDecorationResolver.java

```diff
@@ -227,7 +239,49 @@ protected Object getDecorationAttributeJsonValue(ConfigBag attrs) {
 
         @Override
         protected void addDecorationFromJsonMap(Map<?, ?> decorationJson, List<SpecParameter<?>> decorations) {
-            throw new IllegalStateException("Not called");
+            throw new UnsupportedOperationException("SpecParameterResolver.addDecorationFromJsonMap should never be called.");
+        }
+    }
+
+    public static class TagsResolver extends BrooklynEntityDecorationResolver<Iterable<Object>> {
+        protected TagsResolver(BrooklynYamlTypeInstantiator.Factory instantiator) {
+            super(instantiator);
+        }
+
+        @Override
+        public void decorate(EntitySpec<?> entitySpec, ConfigBag attrs, Set<String> encounteredRegisteredTypeIds) {
+            Iterable<Object> decorationAttributeJsonValue = getDecorationAttributeJsonValue(attrs);
+            if (decorationAttributeJsonValue != null) {
+                entitySpec.tagsAdd(decorationAttributeJsonValue);
+            }
+        }
+
+        @Override
+        protected String getDecorationKind() {
+            return "Brooklyn Tags";
+        }
+
+        @Override
+        protected Iterable<Object> getDecorationAttributeJsonValue(ConfigBag attrs) {
+            Object brooklynTags = attrs.getStringKey(BrooklynCampReservedKeys.BROOKLYN_TAGS);
+            if (brooklynTags == null) {
+                return null;
+            } else if (!(brooklynTags instanceof List)) {
+                throw new IllegalArgumentException(BrooklynCampReservedKeys.BROOKLYN_TAGS + " should be a List of String elements. You supplied " + brooklynTags);
+            } else {
+                checkArgument(Iterables.all((List) brooklynTags, new Predicate() {
+                    @Override
+                    public boolean apply(Object input) {
+                        return !(input instanceof DeferredSupplier) && (input instanceof Serializable);
```

Comment:
That's not the only tag that we are setting. `WrappedEntity` and `ListTag` for example are not `Serializable`. As I said it's arbitrary/not required for the tags to be `Serializable`.